SpatialGDK/Source/SpatialGDK/Private/EngineClasses/Components/SpatialPingComponent.cpp

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "EngineClasses/Components/SpatialPingComponent.h"
+
+#include "GameFramework/PlayerController.h"
+#include "Net/UnrealNetwork.h"
+#include "TimerManager.h"
+#include "Engine/World.h"
+
+USpatialPingComponent::USpatialPingComponent(const FObjectInitializer& ObjectInitializer)
+	: Super(ObjectInitializer)
+{
+	PrimaryComponentTick.bCanEverTick = false;
+	PrimaryComponentTick.bStartWithTickEnabled = false;
+
+	bReplicates = true;
+
+	bStartWithPingEnabled = true;
+}
+
+void USpatialPingComponent::GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const
+{
+	Super::GetLifetimeReplicatedProps(OutLifetimeProps);
+
+	DOREPLIFETIME(USpatialPingComponent, ReplicatedPingID);
+}
+
+void USpatialPingComponent::BeginPlay()
+{
+	Super::BeginPlay();
+	// Attempt to cast the owner of this component to the PlayerController class.
+	// Component will do nothing if cast fails.
+	OwningController = Cast<APlayerController>(GetOwner());
```

Comment:
Should we print something if this fails?

Reply:
Good point, will add a log entry.

Reply:
Log entry added. I considered having the component destroy itself in this scenario but I figure that would be potentially more dangerous than just leaving the component sitting there doing nothing.